src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts

```diff
@@ -347,8 +357,17 @@ export const useSaveAndPublishSetup = (hasPrompt: boolean) => {
       }
 
       if (createdAppletId && ownerId) {
-        navigate(getBuilderAppletUrl(createdAppletId));
-        await dispatch(getAppletWithItems({ ownerId, appletId: createdAppletId }));
+        const oldApplet = getValues();
```

Comment:
pls move this block and the same above into a shared function

Reply:
done